flake-module: fix wrapper for `opentofu` called `terraform`

Enzime opened the pull request:
I removed this option as it should always be equal to `submod.config.terraformWrapper.package.meta.mainProgram`

I couldn't find any uses of it on GitHub: https://github.com/search?utf8=%E2%9C%93&q=path%3A*.nix%20terraformWrapper%20mainProgram%20-is%3Afork&type=code

mdarocha reviewed (approved):
Maybe it'd be good to provide an example in the docs on how to set `mainProgram` of a package in case it's missing (by using overlays for example)

Enzime commented:
I would recommend reviewing the new commits individually. I've updated the description of `terraformWrapper.package` to make it clearer that it should be set to either `pkgs.terraform` or `pkgs.opentofu` (with or without `withPlugins`) meaning that `meta.mainProgram` should always be set.

And I've also added a commit that does a small refactor which removes an unnecessary `mkOption` and `submodule`.

mdarocha reviewed (approved):
Yeah alright, LGTM
